.github/CONTRIBUTING.md

```diff
@@ -8,6 +8,16 @@ The following are certain Google Provider-specific things to be aware of when co
 
 We aim to make the Google Provider a good steward of Go practices. See https://github.com/golang/go/wiki/CodeReviewComments for common Go mistakes that you should attempt to avoid.
 
+## Autogenerated Resources
+
+We maintain 2 different versions of the google terraform provider; the [GA (General Audience) provider](https://github.com/terraform-providers/terraform-provider-google) and the [Beta provider](https://github.com/terraform-providers/terraform-provider-google). We are using code generation tool called Magic Modules that uses the same code base to generate both of these providers. Some Terraform resources are defined in `api.yaml` files [located in magic modules](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/products) and are fully generated, whereas some resources are hand written and located in [the provider/terraform/ folder in magic modules](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/provider/terraform/resources). Generated resources will have a prominent header at the top of the file identifying them.
```

Comment:
It looks like the link for the beta provider is wrong. We should probably link to the Magic Modules repo when introducing it, as well. We should probably also note that the handwritten files are only _temporarily_ handwritten while we convert them to yaml files.